src/ReactiveUI.Maui/RoutedViewHost.cs

```diff
@@ -205,7 +209,13 @@ protected virtual Page PageForViewModel(IRoutableViewModel vm)
 
         ret.ViewModel = vm;
 
-        return (Page)ret;
+        var pg = (Page)ret;
+        MainThread.BeginInvokeOnMainThread(() =>
```

Comment:
This isn't using the inbuilt RxUI way of doing this, so might be better to use RxApp.MainThreadScheduler